parts/linux/cloud-init/artifacts/cse_install.sh

```diff
@@ -420,9 +421,25 @@ extractKubeBinaries() {
         k8s_tgz_tmp="${k8s_downloads_dir}/${k8s_tgz_tmp_filename}"
         mkdir -p ${k8s_downloads_dir}
 
-        retrycmd_get_tarball 120 5 "${k8s_tgz_tmp}" ${kube_binary_url} || exit $ERR_K8S_DOWNLOAD_TIMEOUT
-        if [[ ! -f ${k8s_tgz_tmp} ]]; then
-            exit "$ERR_K8S_DOWNLOAD_TIMEOUT"
+        registry_regex='^.+\/.+\/.+:.+$'
+        if [[ ${kube_binary_url} =~ $registry_regex ]]; then
```

Comment:
I'd recommend using the  BLOCK_OUTBOUND_NETWORK var over a regex check to know if a cluster is NIC

Reply:
Thanks for the comment. Based on API review discussion, bootstrap from cache (acr) is supported even when block_outbound_network=false, so cannot use BLOCK_OUTBOUND_NETWORK here.

Reply:
in fact outside the function, i am using BOOTSTRAP_PROFILE_CONTAINER_REGISTRY_SERVER as the condition to use oras